notifications_utils/statsd_decorators.py

```diff
@@ -32,3 +33,46 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return time_function
+
+
+def statsd_catch(namespace: str, counter_name: str, exception: Type[Exception]):
+    """Increases a statsd counter when a given exception is raised.
+
+    When the expected exception is raised, the statsd counter will be
+    incremented and the initial exception re-raised again.
+
+    When a non-expected exception is raised, no counter increment
+    occurs and nothing is caught: it should go through with no problem.
+
+    All parameters are required.
```

Comment:
Interesting new decorator! Di you find other places in the code where we could use this?

Reply:
Yes, other places in the `validators` file in the *notification-api*. I wanted to introduce it first and see what you think.